src/System.Xml.XDocument/System/Xml/Linq/XDocument.cs

```diff
@@ -350,6 +424,55 @@ public static XDocument Load(XmlReader reader, LoadOptions options)
         {
             if (reader == null) throw new ArgumentNullException("reader");
             if (reader.ReadState == ReadState.Initial) reader.Read();
+
+            XDocument d = InitLoad(reader, options);
+            d.ReadContentFrom(reader, options);
+
+            if (!reader.EOF) throw new InvalidOperationException(SR.InvalidOperation_ExpectedEndOfFile);
+            if (d.Root == null) throw new InvalidOperationException(SR.InvalidOperation_MissingRoot);
+            return d;
+        }
+
+        /// <summary>
+        /// Create a new <see cref="XDocument"/> containing the contents of the
+        /// passed in <see cref="XmlReader"/>.
+        /// </summary>
+        /// <param name="reader">
+        /// An <see cref="XmlReader"/> containing the XML to be read into the new
+        /// <see cref="XDocument"/>.
+        /// </param>
+        /// <param name="options">
+        /// A set of <see cref="LoadOptions"/>.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// A cancellation token.
+        /// </param>
+        /// <returns>
+        /// A new <see cref="XDocument"/> containing the contents of the passed
+        /// in <see cref="XmlReader"/>.
+        /// </returns>
+        public static async Task<XDocument> LoadAsync(XmlReader reader, LoadOptions options, CancellationToken cancellationToken)
+        {
+            if (reader == null) throw new ArgumentNullException("reader");
```

Comment:
It should really be part of its own non-async method that then delegates to an async method for the actual work so that the usage error propagates out synchronously.